Feat/corl 2469 story drawer updates

marcushaddon opened the pull request:
## What does this PR do?
- Adds a "Moderate Story" button to the info drawer
- Removes meta data display
- Change story mode select to checkbox
- Change checkbox wording to match stream config

## What changes to the GraphQL/Database Schema does this PR introduce?
None

## Does this PR introduce any new environment variables or feature flags? 
No.

## How do I test this PR?
Look at https://vmproduct.atlassian.net/browse/CORL-2469 and compare changes.

marcushaddon committed:
remove metadata

marcushaddon committed:
implement changes from Coral Party

marcushaddon committed:
update submit button text

marcushaddon commented:
The buttons mentioned in the ticket (Moderate story and Update) are blue in the stream, but using the same color for the button admin side results in this green color, and it seems like the admin is themed differently.

marcushaddon commented on src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx:
```diff
@@ -0,0 +1,104 @@
+  `);
+
+  const href = useMemo(() => {
+    let link = getModerationLink({ storyID: id });
```
There is a bit of duplicated between this and the stream-side ModerateStreamConainer, but putting that compnent in a shared directory was tricky because of the differing `Local` schema, and so I went with duplicating it for now.

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx:
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable */
```
Why is eslint disabled for this file?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx:
```diff
@@ -0,0 +1,104 @@
+import { ModerateStoryButton_story } from "coral-admin/__generated__/ModerateStoryButton_story.graphql";
+import { ModerateStoryButton_viewer } from "coral-admin/__generated__/ModerateStoryButton_viewer.graphql";
+
+import * as styles from "./ModerateStoryButton.css";
```
Can we do `import styles from...` here for consistency?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/RescrapeStory.tsx:
```diff
@@ -34,7 +34,9 @@ const ScrapeStory: FunctionComponent<Props> = ({ storyID }) => {
         ) : (
-          <Localized id="storyInfoDrawer-triggerRescrape">Rescrape</Localized>
+          <Localized id="storyInfoDrawer-triggerRescrape">
+            Rescrape Story Metadata
```
The issue looks like it just has `Rescrape Metadata` here, so just wanted to check we want it to say `Rescrape Story Metadata`

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StorySettingsContainer.tsx:
```diff
@@ -112,44 +131,19 @@ const StorySettingsContainer: FunctionComponent<Props> = ({
-                      </Localized>
-                    </Flex>
-
                     {/* SAVE/SUBMIT */}
```
Do we want this button to say `Update` instead of `Save` now to match the stream update button? Or we only wanted to update the styles to match?

marcushaddon replied:
Updated but just want to make sure this is what you were referring to

kabeaty replied:
Yep, that's what I was asking about, thanks! One thing I just noticed--it looks like this button is missing localization. Can you please add?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx:
```diff
@@ -0,0 +1,104 @@
+        target="_blank"
+        upperCase
+      >
+        Moderate
```
In the localization file it says `Moderate story`, and here and in the stream it just says `Moderate`. Which do we want for this button?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StoryInfoDrawerContainer.tsx:
```diff
@@ -23,26 +20,16 @@ import StoryStatus from "./StoryStatus";
 export interface Props {
   onClose: () => void;
```
Looks like `onClose` isn't used. Can we remove?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StoryInfoDrawerContainer.tsx:
```diff
@@ -4,17 +4,14 @@ import { graphql } from "relay-runtime";
 
+import ModerateStoryButton from "./ModerateStoryButton";
 import RescrapeStory from "./RescrapeStory";
 import styles from "./StoryInfoDrawerContainer.css";
```
I know this was like this, but do you mind moving the `styles` import down below this section for consistency?

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StoryInfoDrawerQuery.tsx:
```diff
@@ -33,6 +39,10 @@ const StoryInfoDrawerQuery: FunctionComponent<Props> = ({
         }
 
+        if (!props?.viewer) {
+          return <h1>TODO NO VIEWER?!</h1>;
```
It looks like this case still needs to be handled. Maybe we show a spinner here too?

marcushaddon replied:
I changed it to that, but are there cases where viewer is null and it't not because the current user is not logged in? @nick-funk

nick-funk replied:
The viewer can be null because you can access one part of the admin (the login page) and have the viewer be null.

If there is no viewer when you're using the StoryInfoDrawerQuery, it is fine to show a spinner or return null. We don't want to be presenting any of the drawer without a valid viewer.

marcushaddon replied:
aha, good to know

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StorySettingsContainer.tsx:
```diff
@@ -90,13 +88,34 @@ const StorySettingsContainer: FunctionComponent<Props> = ({
+                    <Flex direction="row" className={styles.setting}>
+                      {/* PREMOD ALL COMMENTS ENABLED */}
+                      <Field
+                        name="premoderateComments"
```
Can we make the field name here `moderation` and then use a `parse` and `format` like in `PremodConfig.tsx`, for example, to change to/from a boolean? This would be consistent with how we're handling this in other places, and I think also more readable.

marcushaddon replied:
agreed!

kabeaty reviewed (changes requested):
Looking good overall! I left a few comments/questions throughout.

I also noticed that if someone clicks `Rescrape Metadata` for an archived story, there is an error. Should this (and other configuration settings) be disabled here for archived stories?

marcushaddon commented on src/core/client/admin/components/StoryInfoDrawer/StoryInfoDrawerQuery.tsx:
```diff
@@ -33,6 +39,10 @@ const StoryInfoDrawerQuery: FunctionComponent<Props> = ({
         }
 
+        if (!props?.viewer) {
+          return <h1>TODO NO VIEWER?!</h1>;
```
I think i can assume that there will be a viewer here since Admin is an authenticated view, so I'm going to just assert not null, but let me know if that seems problematic

marcushaddon committed:
remove eslint disable

marcushaddon committed:
address comments

marcushaddon committed:
fix tests

marcushaddon committed:
add post shadow dom TODO

nick-funk commented on src/core/client/admin/components/StoryInfoDrawer/ModerateStoryButton.tsx:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    return link;
+  }, [accessToken, settings, id]);
```
We have this pattern of using `getModerationLink` and then doing the rigamarole of appending an `accessToken` to it.

How do we feel about updating the `getModerationLink` to take an optional `accessToken` parameter so that we can append in this access token more cleanly in one line and not have duplicates of this append logic all over the admin?

marcushaddon replied:
I like that idea. That function does change to a hook in the Shadow DOM PR, and I have a note to update this PR/area of the code once that is merged, maybe it would be easier to do it then?

nick-funk reviewed (changes requested):
Overall, code looks good! Made some small requests and addressed your TODO question around the null viewer.

marcushaddon committed:
hide config options if story archived
show spinner if viewer is null while rendering moderate story button
add todos for after https://github.com/coralproject/talk/pull/3796 is merged

kabeaty commented on src/core/client/admin/components/StoryInfoDrawer/StorySettingsContainer.tsx:
```diff
@@ -57,9 +48,16 @@ const StorySettingsContainer: FunctionComponent<Props> = ({
   const onSubmit = async (
-    values: Partial<StorySettingsContainer_storySettings>
+    values: Partial<StorySettingsContainer_storySettings> & {
+      premoderateComments?: boolean;
```
Can this be removed now that we're not using it for the field anymore?

kabeaty reviewed (approved):
Looks good, thanks for making the updates! I left one more comment, but approved.

marcushaddon committed:
remove unused type

nick-funk reviewed (approved):
Approving as we can make the refactor I suggested later when shadowDOM changes are merged.
